cohorts/model.py

```diff
@@ -0,0 +1,97 @@
+# Copyright (c) 2016. Mount Sinai School of Medicine
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import print_function
+from types import FunctionType
+import numpy as np
+import pandas as pd
+from sklearn.utils import resample
+from sklearn.metrics import roc_auc_score
+import lifelines as ll
+import patsy
+
+def is_single_class(arr, col):
+    if type(arr) == pd.DataFrame:
+        arr_positive_indices = list(arr[arr[col] == 1][col])
+        arr_negative_indices = list(arr[arr[col] == 0][col])
+    else:
+        arr_positive_indices = (arr == 1).nonzero()[0]
+        arr_negative_indices = (arr == 0).nonzero()[0]
+    return len(arr_positive_indices) == 0 or len(arr_negative_indices) == 0
+
+def bootstrap_auc(cohort, func, pred_col="is_benefit", n_bootstrap=1000):
+    """
+    Calculate the boostrapped AUC for a given func trying to predict a pred_col.
+
+    Parameters
+    ----------
+    cohort: cohorts.Cohort
+    func : see cohorts.functions
+    pred_col : the column we're trying to predict
+    n_boostrap : the number of bootstrap samples
+
+    Returns
+    -------
+    list : AUCs for each sampling
+    """
+    scores = np.zeros(n_bootstrap)
+    count_col, df = cohort.as_dataframe(func)
+    preds = df[pred_col].astype(int)
```

Comment:
similar code exists here: https://github.com/hammerlab/cohorts/blob/master/cohorts/plot.py#L152-L159, do you want to refactor the plotting coding to use this as well?